src/Microsoft.DotNet.Web.ProjectTemplates/RazorPagesWeb-CSharp.csproj.in

```diff
@@ -26,7 +26,7 @@
     <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="${MicrosoftAspNetCoreAuthenticationOpenIdConnectPackageVersion}" Condition="'$(OrganizationalAuth)' == 'True' OR '$(IndividualB2CAuth)' == 'True'" />
     <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="${MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion}" Condition="'$(IndividualLocalAuth)' == 'True'" />
     <PackageReference Include="Microsoft.AspNetCore.CookiePolicy" Version="${MicrosoftAspNetCoreCookiePolicyPackageVersion}" />
-    <PackageReference Include="Microsoft.AspNetCore.HttpsPolicy" Version="${MicrosoftAspNetCoreHttpsPolicyPackageVersion}" />
+    <PackageReference Include="Microsoft.AspNetCore.HttpsPolicy" Version="${MicrosoftAspNetCoreHttpsPolicyPackageVersion}" Condition="'$(RequiresHttps)' == 'True' OR '$(NoHttps)' != 'True'"/>
```

Comment:
We want to include the package no matter what so that its easy for customers to turn it on later on if they want to.

Reply:
Do we? I'm ok either way...

Reply:
One possibility would be to have the package reference commented out if it's not being used, with a comment about uncommenting it if the user wants to enable https later. But that might be more trouble than it's worth.

Reply:
Definitely don't want it commented out - it would just look like a bug. I'm OK keeping it the way it is in this PR.

Reply:
A lot of other packages are Condition based as well so I think this keeps that consistency. We do not provide any auth packages by default either.

Reply:
@jbagga OK that makes sense, let's go with that approach in general.